tests/__main__.py

```diff
@@ -5,15 +5,12 @@
 # License: 3-clause BSD.  The full license text is available at:
 #  - https://github.com/uqfoundation/dill/blob/master/LICENSE
 
-from __future__ import print_function
-import glob
-import os
+import glob, os, subprocess as sp
 try:
     import pox
-    python = pox.which_python(version=True, fullpath=False) or 'python'
+    python = pox.which_python(version=True, fullpath=False) or 'python3'
```

Comment:
We have to look this up (it's in one of the UQFoundation's tickets)... but one of (`python`, `python3`) is an alias that isn't guaranteed to be installed on all systems.  So, this change may be fine, and maybe not.

Reply:
This got me a couple times when testing on different Python versions, and it run Python 2 instead (at least I noticed something was wrong, though.

There's a function `shutil.which()` in since Py3.3 that could be used to determine if there's a `python3` command and fall back to `python` as a default. Something like:
```python
default_python = 'python3' if shutil.which('python3') is not None else 'python'
```

Reply:
Here's what I was referring to: https://github.com/uqfoundation/pox/pull/31.  Some environments don't have a `python3`.

Reply:
Are you sure? He says Fedora Linux doesn't have a `python` executable, only `python3`. My suggestion above is almost identical to yours in that issue by the way...

Reply:
whoops. I misread that. :)